### ISSUE TITLE
I tried to upload my site from publii using godaddy server and this is what it shows.  

### ISSUE THREAD

**Issue by nishajose1106** [GetPublii/Publii#1232]
### Operating system

Windows

### Publii version

0.42.1 (build 16233)

### Post editor

None

### Bug description

I tried to upload my site from publii using godaddy server and this is what it shows.  I am able to upload files using filezilla.  But when try the from publii with the same configuration details, this error is what I am getting.  Can anyone please help me? Error.  The application was able to connect with your server but was unable to store files.  Please check file permissions on your server[Bug]:

**Comment by dziudek** [comment]
Hi,

If you are using SFTP, most probably you have provided invalid path in the settings and Publii tries to upload files to the root directory where obviously it has no permissions to write

**Comment by nishajose1106** [comment]
Thank you so much for your reply.  Could you please find  the files attached and guide me further if I am wrong anywhere?
![Publii server connection page Error](https://github.com/GetPublii/Publii/assets/140804105/428a639b-d65c-4cee-b70d-3b4751522848)
![Publii server connection page](https://github.com/GetPublii/Publii/assets/140804105/50b597da-5b64-4562-a2c4-db835e2807b9)

**Comment by nishajose1106** [comment]
Thank you so much for replying to me.  Can you please look into the attached files and guide me further if I am wrong anywhere?
![Publii server connection page Error](https://github.com/GetPublii/Publii/assets/140804105/0906db21-3290-468b-97d3-a9ca993be619)
![Publii server connection page](https://github.com/GetPublii/Publii/assets/140804105/08ef236b-7f29-48db-abb9-23eb0ebb9000)

**Comment by dziudek** [comment]
1. Please change password to your server because everybody who subscribes issues on this repository got your password from screen in your previous message 😅

2. In my opinion the path is not absolute - maybe it is a reason. You can also try different combinations, because most probably the path is an issue in your case

**Comment by Pakeydev** [comment]
Exact same issue. Login and uploading using the same credentials is working well with Filezilla.

> 2. In my opinion the path is not absolute - maybe it is a reason. You can also try different combinations, because most probably the path is an issue in your case

I have tried all possible combination but it still failed.

**Comment by dziudek** [comment]
@EKliptus - it can be some incompatibility of the our FTP library and your server. In next few days I will prepare Publii version which allows to use other FTP library - do you want to test it then? I can send you such experimental build when it will be ready

**Comment by jfjensen** [comment]
I have the same issue. I'm trying to upload my blog to an FTP server running on port 21 via CPanel.

**Comment by jfjensen** [comment]
In the logs, it says: "Error: You cannot use PASV on IPv6 connections. Use EPSV instead."

**Comment by dziudek** [comment]
@jfjensen - in Publii v.0.45 we will have an option to use alternative FTP library which should have better support for IPv6 - I do hope that it will solve your issue. I will let you know when this version will be available

**Comment by jfjensen** [comment]
@dziudek I have just updated Publii to v.0.45.0 but the log still shows the same error. These are the last lines of `deployment-process.log` after an attempt to sync the website:
[Tue, 05 Mar 2024 13:29:14 GMT] [connection] > 'PASV'
[Tue, 05 Mar 2024 13:29:14 GMT] [connection] < '425 You cannot use PASV on IPv6 connections. Use EPSV instead.\r\n'
[Tue, 05 Mar 2024 13:29:14 GMT] [parser] < '425 You cannot use PASV on IPv6 connections. Use EPSV instead.\r\n'
[Tue, 05 Mar 2024 13:29:14 GMT] [parser] Response: code=425, buffer='You cannot use PASV on IPv6 connections. Use EPSV instead.'
[Tue, 05 Mar 2024 13:29:14 GMT] -> files.publii.json
[Tue, 05 Mar 2024 13:29:14 GMT] Error: You cannot use PASV on IPv6 connections. Use EPSV instead.
[Tue, 05 Mar 2024 13:29:14 GMT] [connection] > 'QUIT'
[Tue, 05 Mar 2024 13:29:14 GMT] [connection] < '221-Goodbye. You uploaded 0 and downloaded 0 kbytes.\r\n221 Logout.\r\n'
[Tue, 05 Mar 2024 13:29:14 GMT] [parser] < '221-Goodbye. You uploaded 0 and downloaded 0 kbytes.\r\n221 Logout.\r\n'
[Tue, 05 Mar 2024 13:29:14 GMT] [parser] Response: code=221, buffer='Goodbye. You uploaded 0 and downloaded 0 kbytes.\r\nLogout.'
[Tue, 05 Mar 2024 13:29:14 GMT] FTP CONNECTION CLOSED: false

**Comment by dziudek** [comment]
@jfjensen - you has been faster with your tests than my notification for you about the update :)

Please go to the Application Settings and under experimental features section, please select "Use alternative FTP library for deployment" and then please try again

We did not enabled it by default due backward compatibility

**Comment by jfjensen** [comment]
@dziudek I'm sorry but I'm still getting the same error. I have now uploaded the files to my website by using FileZilla instead.

**Comment by Uzayer** [comment]
Same issue over here. Just opened another thread before seeing this one but yes.

**Comment by dziudek** [comment]
@jfjensen - but do you have the same error under logs? That's weird, because in theory the alternative FTP library supports IPv6

**Comment by jfjensen** [comment]
@dziudek I have turned on the alternative FTP library. In the log I still get this:

[Thu, 07 Mar 2024 17:12:57 GMT] [connection] > 'TYPE I'
[Thu, 07 Mar 2024 17:12:57 GMT] [connection] < '200 TYPE is now 8-bit binary\r\n'
[Thu, 07 Mar 2024 17:12:57 GMT] [parser] < '200 TYPE is now 8-bit binary\r\n'
[Thu, 07 Mar 2024 17:12:57 GMT] [parser] Response: code=200, buffer='TYPE is now 8-bit binary'
[Thu, 07 Mar 2024 17:12:57 GMT] [connection] > 'PASV'
[Thu, 07 Mar 2024 17:12:57 GMT] [connection] < '425 You cannot use PASV on IPv6 connections. Use EPSV instead.\r\n'
[Thu, 07 Mar 2024 17:12:57 GMT] [parser] < '425 You cannot use PASV on IPv6 connections. Use EPSV instead.\r\n'
[Thu, 07 Mar 2024 17:12:57 GMT] [parser] Response: code=425, buffer='You cannot use PASV on IPv6 connections. Use EPSV instead.'
[Thu, 07 Mar 2024 17:12:57 GMT] <- files.publii.json
[Thu, 07 Mar 2024 17:12:57 GMT] (!) ERROR WHILE DOWNLOADING files-remote.json
[Thu, 07 Mar 2024 17:12:57 GMT] Error: You cannot use PASV on IPv6 connections. Use EPSV instead.

**Comment by dziudek** [comment]
@jfjensen - are you sure that you have enabled the alternative FTP library?

I am asking, because logs from this library looks totally different - lets look at my example:

```
[Sat, 09 Mar 2024 08:28:53 GMT] > TYPE I
[Sat, 09 Mar 2024 08:28:53 GMT] < 200 TYPE is now 8-bit binary

[Sat, 09 Mar 2024 08:28:53 GMT] > STRU F
[Sat, 09 Mar 2024 08:28:53 GMT] < 200 F OK

[Sat, 09 Mar 2024 08:28:53 GMT] > OPTS UTF8 ON
[Sat, 09 Mar 2024 08:28:53 GMT] < 504 Unknown command

[Sat, 09 Mar 2024 08:28:53 GMT] > OPTS MLST type;size;modify;unique;unix.mode;unix.owner;unix.group;unix.ownername;unix.groupname;
[Sat, 09 Mar 2024 08:28:53 GMT] < 200  MLST OPTS type;size;sizd;modify;UNIX.mode;UNIX.uid;UNIX.gid;unique;

[Sat, 09 Mar 2024 08:28:53 GMT] > PBSZ 0
[Sat, 09 Mar 2024 08:28:53 GMT] < 200 PBSZ=0

[Sat, 09 Mar 2024 08:28:53 GMT] > PROT P
[Sat, 09 Mar 2024 08:28:53 GMT] < 200 Data protection level set to "private"

[Sat, 09 Mar 2024 08:28:53 GMT] Trying to find optimal transfer strategy...
[Sat, 09 Mar 2024 08:28:53 GMT] > EPSV
[Sat, 09 Mar 2024 08:28:53 GMT] < 500 Unknown command

[Sat, 09 Mar 2024 08:28:53 GMT] > PASV
[Sat, 09 Mar 2024 08:28:53 GMT] < 227 Entering Passive Mode (91,211,222,125,235,64)
```

As you can see there are no `[connection]` and `[parser]` strings there :)

**Comment by jfjensen** [comment]
@dziudek Yes, I even tried to turn this feature Off and then On again, but that made no difference.

![Screenshot 2024-03-09 100308](https://github.com/GetPublii/Publii/assets/2771076/ca01b102-542d-4dd9-ba2b-1f4758938b07)

**Comment by dziudek** [comment]
@jfjensen - please make sure that under Documents/Publii/config/app-config.json you have fragment:

```
"experimentalFeatureAppFtpAlt": true,
```

If yes, then it is almost impossible that Publii uses old library instead of the new one 😳

**Comment by jfjensen** [comment]
@dziudek I checked and it is indeed set to true. It still does not work. I even tried to uninstall and re-install but this did not solve the issue either.

**Comment by dziudek** [comment]
@jfjensen - sorry, but at this moment I have no other ideas why the new FTP library is not working for you - the code which enables use of it is quite simple and if the app config contains the proper flag then it should work as expected. Maybe if somebody else will have similiar issue, then we will be able to find a reason :(
